lpcompiler.pas

```diff
@@ -2731,12 +2754,15 @@ function TLapeCompiler.ParseFor(ExprEnd: EParserTokenSet = ParserToken_Expressio
       end
     else
     begin
-      Result.Counter := ParseExpression();
-      Expect([tk_kw_To, tk_kw_DownTo], False, True);
+      Result.Counter := ParseExpression([tk_op_In]);
+      Expect([tk_op_In, tk_kw_To, tk_kw_DownTo], False, True);
     end;
 
     if (Tokenizer.LastTok = tk_kw_DownTo) then
-      Result.WalkDown := True;
+      Result.WalkDown := True
+    else if (Tokenizer.LastTok = tk_op_In) then
+      Result.WalkIn := True;
```

Comment:
Rather than two booleans, maybe an enum value? The same goes for IsOperator in ParseMethodHeader.